mypy/plugin.py

```diff
@@ -263,17 +264,26 @@ def typed_dict_get_signature_callback(ctx: MethodSigContext) -> CallableType:
             and len(ctx.args[0]) == 1
             and isinstance(ctx.args[0][0], StrExpr)
             and len(signature.arg_types) == 2
-            and len(signature.variables) == 1):
+            and len(signature.variables) == 1
+            and len(ctx.args[1]) == 1):
         key = ctx.args[0][0].value
         value_type = ctx.type.items.get(key)
+        ret_type = signature.ret_type
         if value_type:
+            default_arg = ctx.args[1][0]
+            if (isinstance(value_type, TypedDictType)
+                    and isinstance(default_arg, DictExpr)
+                    and len(default_arg.items) == 0):
+                # Caller has empty dict {} as default for typed dict.
```

Comment:
Why do we have to special-case this? Assuming the value type is some TypedDict, and the default passed to get() is some dict literal, isn't the natural union type resulting from the two an appropriate TypedDict? Even if it isn't, shouldn't we use the value type as a context for inferring the type of the dict literal? ISTM that `d.get('x', {'y': 1})` ought to work too.